File: examples/android/CMeSdkJava/lib/src/main/java/msdk/java/messages/StateMachineState.java
package msdk.java.messages;

public enum StateMachineState {
    NONE(0),
    INITIALIZED(1),
    OFFER_SENT(2),
    REQUEST_RECEIVED(3),
    ACCEPTED(4),
    UNFULFILLED(5),
    EXPIRED(6),
    REVOKED(7),
    REDIRECTED(8),
    REJECTED(9);

    private final int value;

    StateMachineState(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    public boolean matches(int state) {
        return this.value == state;
    }
}

File: examples/android/MSDKSampleAppKotlin/app/src/main/java/msdk/kotlin/sample/types/StateMachineState.kt
package msdk.kotlin.sample.types

enum class StateMachineState(val value: Int) {
    NONE(0), INITIALIZED(1), OFFER_SENT(2), REQUEST_RECEIVED(3), ACCEPTED(4), UNFULFILLED(5), EXPIRED(
        6
    ),
    REVOKED(7), REDIRECTED(8), REJECTED(9);

    fun matches(state: Int): Boolean {
        return value == state
    }

}